invisible_cities/reco/deconv_functions.py

```diff
@@ -4,16 +4,78 @@
 from typing  import List
 from typing  import Tuple
 from typing  import Callable
+from typing import Optional
 
 from scipy                  import interpolate
 from scipy.signal           import fftconvolve
 from scipy.signal           import convolve
 from scipy.spatial.distance import cdist
+from scipy import ndimage as ndi
 
 from ..core .core_functions import shift_to_bin_centers
 from ..core .core_functions import in_range
 
 from .. types.symbols       import InterpolationMethod
+from .. types.symbols       import CutType
+
+
+
+import warnings
+
+
+def isolate_satellites(df : np.ndarray, dist : int = 2, size : int = 10) -> np.ndarray:
+    '''
+    An adaptation to the scikit-image (v0.24.0) function below:
+    https://github.com/scikit-image/scikit-image/blob/main/skimage/morphology/misc.py#L59-L151
+    
+    Identifies satellite energy depositions within deconvolution image by size
+    and proximity to other depositions.
+
+    In practice, input array is a set of boolean 'pixels' that it then categorises as 
+    satellite or non-satellite bundles based on given parameters.
+    
+    Parameters
+    ----------
+    df    : masked array containing pixels above energy cut
+    dist  : Maximum distance between two pixels to be considered part of the same deposit.
+    size  : Minimum number of pixels per deposit under which it will be labelled as a satellite deposit.
+
+    Returns
+    ----------
+    array : masked array of all satellite deposits
+
+    
+    '''
+    try:
+        array = data.copy().astype(bool)
+    except:
+        print("Please ensure the array passed through is boolean compatible.")
+    if size == 0:
+        warning.warn(f'Satellite size set to zero. No satellites will be removed')
+        return array # will cause the result to be identical to the input
```

Comment:
Apologies, this is due to a typo in the commit. I altered the input np.ndarray to be `df`, rather than `data`. This will be fixed in the next commit.

But yes, it returned the input as the output.